qiskit/pulse/transforms/channel_transforms.py

```diff
@@ -0,0 +1,86 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+import numpy as np
+from typing import Union
+
+from base_transforms import target_qobj_transform
+from qiskit.providers.ibmq import IBMQBackend
+from qiskit.pulse import Schedule, Play, MeasureChannel
+from qiskit.pulse.channels import PulseChannel
+from qiskit.visualization.pulse_v2.events import ChannelEvents
+
+def get_channel_waveform(sched: Schedule, 
+                         chan: PulseChannel, 
+                         backend: Union[None, IBMQBackend] = None,
+                         qubit_index: Union[None, int] = None,
+                         chan_freq: Union[None, float] = None,
+                         dt: float = 2e-9 / 9, 
+                         apply_carrier_wave: bool = False):
+    """Returns the waveforms on a PulseChannel.
+    
+    Args:
+        sched: The pulse Schedule object.
+        chan: The PulseChannel on which the waveform is to be returned.
+        backend: An IBMQBackend from which the qubit frequency and dt 
+            are to be extracted.
+        qubit_index: An integer indicating the qubit index.
+        chan_freq: A float indicating the channel wave frequency. Not necessary if 
+            both backend and qubit_index are specified.
+        dt: Qubit drive channel timestep in seconds. Default to the 2/9 ns.
+        apply_carrier_wave: Whether the carrier wave is applied to the waveforms.
+        
+    Returns:
+        chan_waveform: A complex-valued array of the waveform on the 
+            given PulseChannel.
+    """
+    # Check consistency of arguments
+    if not isinstance(chan, PulseChannel):
+        raise TypeError("The channel must be a DriveChannel, " 
+                        "ControlChannel or a MeasureChannel")
+    if apply_carrier_wave:
+        if backend is not None and qubit_index is not None:

```

Comment:
Priority should be to use `chan_freq` if supplied as override and fall back to the `backend` if `chan_freq` is not supplied.

Reply:
Ok I will address this in the argument checking section